### PR TITLE
Refs #10741 - Support for smart_proxy_dynflow

### PR DESCRIPTION
**Pull request by iNecas**
Ability to trigger action on foreman server and a corresponding action in
smart proxy, get the callback at the end and dispatch the control events (such
as cancel)

Depends on https://github.com/theforeman/smart_proxy_dynflow/pull/1

### CODE REVIEW

**Comment by iNecas** [comment]
[test]

**Comment by iNecas** [comment]
I've rebased and fixed tasks reloading for planned tasks

**Comment by iNecas** [comment]
Due to cancelling enhancements, it requires https://github.com/Dynflow/dynflow/pull/162

File: lib/foreman_tasks/test_extensions.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module ForemanTasks
+  module TestExtensions
+    module AccessPermissionsTestExtension
+      def setup
+        super
+        if defined?(AccessPermissionsTest) && self.class == AccessPermissionsTest
+          skip if __name__.include?('foreman_tasks/api/tasks/callback')
```
Comment (ares) [inline lib/foreman_tasks/test_extensions.rb]: would be nice to add justifying comment here

**Comment by ares** [comment]
ACK, comment is optional :-)

**Comment by iNecas** [comment]
The jenkins seemed to get stuck on some platforms, but majority of them worked: I don't see anything related ot this code: merging so that we can move forward with https://github.com/theforeman/foreman_remote_execution/pull/11

**Pull request merged**